CONTRIBUTING/CONTRIBUTING.md

```diff
@@ -129,20 +138,60 @@ Cli follows the python [`pep8`](https://peps.python.org/pep-0008/) coding style.
 
 We use [pre-commit](https://pre-commit.com/) to enforce coding style using [`black`](https://github.com/psf/black), and [`isort`](https://pycqa.github.io/isort/).
 
-You can invoke formatting with:
+You can invoke formatting with (use `--` followed by a single file):
 
 ```shell
 tox -e ruff
 ```
 
 In addition, we use [`pylint`](https://www.pylint.org) to perform static code analysis of the code.
 
-You can invoke the linting with the following command
+You can invoke the linting with the following command (use `--` followed by a single file):
 
 ```shell
 tox -e lint
 ```
 
+### Type Checking
```

Comment:
It seems like all of the entries in this section follow the same format of tool link, description, and an example of how it's invoked. Rather than having an independent section for each, would it make sense to have a list or a table for each?

Maybe something like:

```
- Type checking: We use [`mypy`](https://mypy.readthedocs.io/en/stable/) to perform type-checking: `tox -e mypy`
- etc.
```

Or you could also have a table for this:

```
| Purpose | Tool name | Description | Example |
| - | - | - | - |
| Type-checking | [mypy](https://mypy.readthedocs.io/en/stable/) | Performs static type analysis across the project to catch bugs | `tox -e mypy` |
```

Which would render such as:
| Purpose | Tool name | Description | Example |
| - | - | - | - |
| Type-checking | [mypy](https://mypy.readthedocs.io/en/stable/) | Performs static type analysis across the project to catch bugs | `tox -e mypy` |